content/blog/2020-08-05-cml-self-hosted-runners-on-demand-with-gpus.md

```diff
@@ -0,0 +1,181 @@
+---
+title: CML self-hosted runners on demand with GPUs
+date: 2020-08-05
+description: |
+  Bring reproducibility and GPUs to your ML projects in Github or Gitlab pipelines with CML self-hosted runners.
+descriptionLong: |
+  When we train our models we usually need special hardware like big memory or GPUs. 
+  How can we provide our CI/CD pipeline with such hardware?
+  Even more, how could we ensure the reproducibility of the experiment? 
+  Find out how to set up your own self-hosted runners on demand with GPUs for fast training and ensuring reproducibility in this blog post!
+picture: 2020-08-05/header.png
+author: david_g_ortega
+commentsUrl: https://discuss.dvc.org/t/cml-self-hosted-runners-on-demand-with-gpus/462
+tags:
+  - CML
+  - Continuous Machine Learning
+  - GPUs
+  - Self-hosted Runner
+  - Reproducibility
+  - Tutorial
+---
+
+When setting your CI/CD workflow for your machine learning (ML) project you
+might find that not Github nor Gitlab provides certain computing capabilities
+like GPU, high memory, big processors... capabilities that really matter in some
+machine learning (ML) problems.
+
+To overcome this hardware hurdle, one practical approach is to use self-hosted
+runners: runners that you manage, but are accessible to your CI/CD system for
+executing jobs. It could be an EC2 instance or the GPU under your desk. In our
+recently-released project, Continuous Machine Learning (CML), we’ve introduced a
+tool called cloud-runner. It's a docker container that acts as a thin wrapper
+over GitLab and GitHub runners, adding some extra capabilities.
+
+Benefits of using CML cloud runner:
```

Comment:
For grammar, I'd rewrite:

`Here are some benefits of using CML cloud-runner:`

Reply:
Also- are we still calling it cloud-runner since the cloud-runner docker image has been merged into base cml?

Reply:
> Also- are we still calling it cloud-runner since the cloud-runner docker image has been merged into base cml?

I thought the same, however, it could be like and internal name? I mean something that we know it exists...

Reply:
```Here are some benefits of using CML as a self-hosted runner:```